File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        Set<Integer> hashset = new HashSet<>();
        for (int i : nums) { 
            if (hashset.contains(i)) {
                return true;
            }
            hashset.add(i);
        }
        return false;
        
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        # [4, 5, 6, 1, 2, 3]

        l, r = 0, len(nums) - 1

        while l < r:
            mid = l + (r - l )// 2
            
            if nums[mid] < nums[r]:
                r = mid
            else:
                l = mid + 1
        return nums[l]
